_posts/2018-02-21-symfony-route-test.md

```diff
@@ -0,0 +1,112 @@
+categories:
+- Symfony
+---
+<b>How much faster is the new Symfony router in a real world application?</b>
+
+<b>We ran some tests to find out.</b>
+
+## An improved router
+
+A few days ago we received the fantastic news that the Symfony router had been improved significantly. Of course we wanted to see how
+the changes would affect our application, so we decided to test it against our own routes. 
+
+If you somehow missed the news you can read the official announcement here: 
+<https://symfony.com/blog/new-in-symfony-4-1-fastest-php-router>
+
+If you want to know how the new router works, read Nicolas Grekas [post](https://medium.com/@nicolas.grekas/making-symfonys-router-77-7x-faster-1-2-958e3754f0e1) where he dives more into detail about the new changes.  
+
+## Our setup 
+* PHP version: 7.1.13
+* Static routes: 285
+* Dynamic routes: 350
+
+## The test process 
+The process in itself is quite simple. Since the performance of the matcher is based on the amount of configured routes that we have, 
+we want to test against the first route, some random routes, and the last route, to get a realistic test case. 
+This was done for both the static and the dynamic routes. 
+
+{% highlight php%}
+class RouteTestCommand extends BaseCommand
+{
+    const NR_OF_MATCHES = 50000;
+    protected static $defaultName = 'route:test';
+    private $router;
+
+    public function __construct(RouterInterface $router)
+    {
+        parent::__construct();
+        $this->router = $router;
+    }
+    
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $output->writeln('First static route: '.$this->matchRoutes(['/admin/statistics/index']));        
+        // Add random/last static routes..
+
+        $output->writeln('First dynamic route: '.$this->matchRoutes(['/admin/statistics/3aa63971-1f61-4b28-bd60-5bb1c9582c8a/show']));
+        // Add random/last dynamic routes..
+        
+        $output->writeln('Not Found: '.$this->matchRoutes(['/foo/bar/baz']));
+    }
+
+    private function matchRoutes(array $routes): float
+    {
+        $startTime = microtime(true);
+        foreach ($routes as $route) {
+            for ($i = 0; $i < self::NR_OF_MATCHES; ++$i) {
+                $this->router->match($route);
+            }
+        }
+        $time = microtime(true) - $startTime;
+
+        return number_format(1000 * $time / count($routes), 2);
+    }
+{% endhighlight %}
+
+### Symfony 3.4
+In Symfony 3.4, the matcher has to iterate through all/most of the configured routes and try to find the route for the given url. 
```

Comment:
it doesn't actually iterate through all of them; instead, it checks groups routes by same static prefix, then check the subset one by one (can be recursive)
dunno how to formulate sorry :)

Reply:
I know. That is why the `/most` part is there. We will try to reformulate.